File: PCM_Mobile/lib/models/wallet_transaction.dart
import 'enums.dart';

class WalletTransaction {
  final int id;
  final int memberId;
  final double amount;
  final TransactionType type;
  final TransactionStatus status;
  final String? description;
  final String? proofImageUrl;
  final String? relatedId;
  final DateTime createdDate;
  final DateTime? approvedDate;
  final String? approvedBy;

  WalletTransaction({
    required this.id,
    required this.memberId,
    required this.amount,
    required this.type,
    required this.status,
    this.description,
    this.proofImageUrl,
    this.relatedId,
    required this.createdDate,
    this.approvedDate,
    this.approvedBy,
  });

  factory WalletTransaction.fromJson(Map<String, dynamic> json) {
    return WalletTransaction(
      id: json['id'],
      memberId: json['memberId'],
      amount: (json['amount'] as num).toDouble(),
      type: TransactionType.values[json['type']],
      status: TransactionStatus.values[json['status']],
      description: json['description'],
      proofImageUrl: json['proofImageUrl'],
      relatedId: json['relatedId'],
      createdDate: DateTime.parse(json['createdDate']),
      approvedDate: json['approvedDate'] != null
          ? DateTime.parse(json['approvedDate'])
          : null,
      approvedBy: json['approvedBy'],
    );
  }
}

File: PCM_Backend/PCM.API/Models/WalletTransaction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PCM.API.Models;

[Table("519_WalletTransactions")]
public class WalletTransaction
{
    [Key]
    public int Id { get; set; }
    
    public int MemberId { get; set; }
    
    [Column(TypeName = "decimal(18,2)")]
    public decimal Amount { get; set; }
    
    public TransactionType Type { get; set; }
    
    public TransactionStatus Status { get; set; } = TransactionStatus.Pending;
    
    [MaxLength(100)]
    public string? RelatedId { get; set; }
    
    [MaxLength(500)]
    public string Description { get; set; } = string.Empty;
    
    public DateTime CreatedDate { get; set; } = DateTime.Now;
    
    [MaxLength(500)]
    public string? ProofImageUrl { get; set; }
    
    // Navigation
    [ForeignKey(nameof(MemberId))]
    public virtual Member Member { get; set; } = null!;
}
